source/blog/2017-07-18-running-kubernetes-on-fedora-atomic-26.html.md

```diff
@@ -0,0 +1,127 @@
+---
+title: Running Kubernetes on Fedora Atomic 26
+author: jbrooks
+date: 2017-07-18 23:12:57 UTC
+tags: kubernetes, ansible, kubeadm
+comments: true
+published: true
+---
+
+Fedora Atomic 26 relies primarily on Kubernetes for automating deployment, scaling, and operations of application containers across clusters of hosts. Below is an overview of your options for installing and configuring Kubernetes clusters on Fedora Atomic hosts.
+
+# Install Kubernetes
+
+## Use Built-in Packages
+
+Fedora Atomic 26 ships with Kubernetes packages baked into the system image. The specific version of Kubernetes included matches the latest release marked stable for f26 in Fedora's [updates system](https://bodhi.fedoraproject.org/updates/?packages=kubernetes&release=F26). If this is the version you wish to run, you can move on to the Manual Deployment, Ansible Deployment or Kubeadm Deployment sections.
+
+### Updates and Testing Packages
+ 
+If there is a newer stable Kubernetes version available that hasn't yet appeared in a two-weekly Fedora Atomic release, you can access it by rebasing to the updates ref of Fedora Atomic, which is recomposed each night to track the latest stable packages:
+
+```
+rpm-ostree rebase fedora-atomic:fedora/26/x86_64/updates/atomic-host -r
+```
+
+Similarly, if there is a newer Kubernetes version available in Fedora's updates-testing repository, you can access it by rebasing to the testing ref of Fedora Atomic, which is recomposed each night to track the latest testing packages:
+
+```
+rpm-ostree rebase fedora-atomic:fedora/26/x86_64/testing/atomic-host -r
+```
+
+## Use System Containers
+
+You can install and run versions of Kubernetes packaged for different Fedora releases using [system containers](http://www.projectatomic.io/blog/2016/09/intro-to-system-containers/). For instance, you could run the older Fedora 25 version (currently 1.5.3) or the newer rawhide version (currently 1.7.1) on your Fedora Atomic 26 host. System containers place systemd unit files in `/etc/systemd/system`, where they override the unit files from the packages baked into the image.
+
+_note: this section below needs registry work. Are the f25 containers still being updated? Will the rawhide containers be updated? I'll probably point to my personal docker namespace, but it'd be better if all of these were available on Fedora's_
+
+### Run on your kubernetes master:
+
+```
+# atomic install --system --name kube-apiserver docker.io/jasonbrooks/kubernetes-apiserver:rawhide
```

Comment:
I think the reason to do that is because we're removing baked kube in the future so running it containerized is the way forward and you should start doing that.